src/cdk/tree/nested-node.ts

```diff
@@ -81,8 +82,10 @@ export class CdkNestedTreeNode<T> extends CdkTreeNode<T> implements AfterContent
       childrenNodes.pipe(takeUntil(this._destroyed))
         .subscribe(result => this.updateChildrenNodes(result));
     }
-    this.nodeOutlet.changes.pipe(takeUntil(this._destroyed))
-        .subscribe(() => this.updateChildrenNodes());
+  }
+
+  ngAfterContentChecked() {
```

Comment:
We should be careful with this since it'll run on every change detection. Can we get away with doing a diff ourselves and only re-rendering if the outlet is different?